subworkflows/local/mtx_conversion.nf

```diff
@@ -2,21 +2,44 @@
 include { MTX_TO_H5AD   }             from '../../modules/local/mtx_to_h5ad.nf'
 include { CONCAT_H5AD   }             from '../../modules/local/concat_h5ad.nf'
 include { MTX_TO_SEURAT }             from '../../modules/local/mtx_to_seurat.nf'
+include { H5AD_TO_10X   }             from '../../modules/local/h5ad_to_10x.nf'
+include { GENE_MAP      }             from '../../modules/local/gene_map.nf'
+include { KALLISTOBUSTOOLS_REF }        from '../../modules/nf-core/kallistobustools/ref/main'
 
 workflow MTX_CONVERSION {
 
     take:
     mtx_matrices
     samplesheet
+    gtf
+    txp2gene
+    genome_fasta
+    output_10x
+
 
     main:
         ch_versions = Channel.empty()
         // Convert matrix do h5ad
         //
+
         MTX_TO_H5AD (
             mtx_matrices
         )
 
+        if (output_10x) {
+            // Generate the t2g file to enrich the 10x files with gene names
+            KALLISTOBUSTOOLS_REF( genome_fasta, gtf, 'standard' )
+            txp2gene = KALLISTOBUSTOOLS_REF.out.t2g.collect()
```

Comment:
is this not available somewhere already? Sometimes this is even provided as input file I think.